docs: update description of useFragment hook (#10638)
apollographql/apollo-client: docs/source/api/react/hooks-experimental.mdx

### Using `useFragment_experimental`

> > A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) defines a subselection of fields on a GraphQL type, that can help define component-level data dependencies or allow re-use between multiple queries and mutations. [See the API reference.](../../data/fragments)

Reviewer: Small tweak: our [fragment docs](https://www.apollographql.com/docs/react/data/fragments/) use the term "subset":

```suggestion
> A [GraphQL fragment](http://graphql.org/learn/queries/#fragments) defines a subset of fields on a GraphQL type that can be used to specify component-level data dependencies or allow re-use between multiple queries and mutations. [See the API reference.](../../data/fragments)
```

Author: I love that, ty! Updated 👍
